pkg/handlers/user.go

```diff
@@ -110,8 +119,14 @@ func (h *UserHandler) GetVariation(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	experimentKey := chi.URLParam(r, "experimentKey")
-	renderVariation(w, r, experimentKey, false, optlyClient, optlyContext)
+	experiment, err := middleware.GetExperiment(r)
+	if err != nil {
+		middleware.GetLogger(r).Error().Err(err).Msg("Calling middleware GetExperiment")
+		RenderError(err, http.StatusInternalServerError, w, r)
```

Comment:
We have been returning a `http.StatusUnprocessableEntity` (422) if the context wasn't properly populated. (I think I stole that response from a middleware example) Here we're returning a 500. 500 makes sense since if we got here it's probably due to a mis configuration on the server side. Can you make the others return a 500 as well for consistency?